Disable hl tools by default (#2313)
HDFGroup/hdf5: release_docs/RELEASE.txt

    Configuration
    -------------

>     - Change the default for building the high-level tools

Reviewer: I wonder if it might be good to call out which tools are getting disabled here by default since I imagine some folks would want to know that building of h5watch is opt-in after this.

Reviewer 2: Agree. The tools are gif2hdf and hdf2gif as well as h5watch

Author: done

>       The gif2hdf5, hdf2gif and h5watch high-level tools are deprecated and will be removed

Reviewer: Is h5watch really deprecated?

Author: All three tools get disabled.

Author: I could split the line to make h5watch in a sentence by itself. A bit confusing though.

    Configuration
    -------------
    - Change the default for building the high-level tools

>       The gif2hdf5 and hdf2gif high-level tools are deprecated and will be removed

Reviewer 3: ```suggestion
      The gif2h5 and h52gif high-level tools are deprecated and will be removed
```

>       in a future release. The default build setting for them have been changed from enabled

Reviewer 3: ```suggestion
      in a future release. The default build setting for them has been changed from enabled
```

      to disabled. A user can enable the build of these tools if needed.
          autotools:   --enable-hlgiftools
          cmake:       HDF5_BUILD_HL_GIF_TOOLS=ON

> 

Reviewer 3: ```suggestion
The gif2h5 and h52gif tools were previously enabled or disabled by the --enable-hltools / HDF5_BUILD_HL_TOOLS option.  That option also controlled building and installing the h5watch tool, which will continue to be built and installed by default but may be disabled using that option.
```
